<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NATRuleOptionsDialog_q</class>
 <widget class="QWidget" name="NATRuleOptionsDialog_q">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>942</width>
    <height>345</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NAT Rule Options</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QStackedWidget" name="widgetStack">
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="WStackPage">
      <layout class="QVBoxLayout">
       <property name="spacing">
        <number>12</number>
       </property>
       <property name="margin">
        <number>12</number>
       </property>
       <item>
        <widget class="QFrame" name="frame1">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout">
          <property name="margin">
           <number>2</number>
          </property>
          <property name="spacing">
           <number>2</number>
          </property>
          <item row="2" column="0" colspan="3">
           <widget class="QLabel" name="hint_no_go">
            <property name="text">
             <string>No options are available for this firewall platform</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <spacer>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="iptNATRuleOptions">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Normally, if object used to define translated source address is an interface marked as &quot;dynamic&quot;,  fwbuilder generates NAT rule with target MASQUERADE. However masquerading has problems with policy routing. This option makes fwbuilder use SNAT target instead.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QCheckBox" name="ipt_use_snat_instead_of_masq">
         <property name="text">
          <string>Use SNAT target instead of MASQUERADING for NAT rules with dynamic interfaces</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QCheckBox" name="ipt_use_masq">
         <property name="text">
          <string>Always use MASQUERADE target instead of SNAT</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QCheckBox" name="ipt_nat_random">
         <property name="text">
          <string>Randomize port mapping (translates to --random)</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="2">
        <widget class="QCheckBox" name="ipt_nat_persistent">
         <property name="text">
          <string>Give a client the same source-/destination-address for each connection (translates to --persistent, requires iptables 1.4.3 or later).</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>81</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pfNATRuleOptions">
      <layout class="QGridLayout">
       <property name="margin">
        <number>12</number>
       </property>
       <property name="spacing">
        <number>12</number>
       </property>
       <item row="3" column="0">
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0" rowspan="3">
        <widget class="QGroupBox" name="buttonGroup1">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Pool type</string>
         </property>
         <layout class="QGridLayout">
          <property name="margin">
           <number>12</number>
          </property>
          <property name="spacing">
           <number>12</number>
          </property>
          <item row="0" column="0">
           <widget class="QRadioButton" name="pf_pool_type_none">
            <property name="text">
             <string>default</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QRadioButton" name="pf_bitmask">
            <property name="text">
             <string>bitmask</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QRadioButton" name="pf_random">
            <property name="text">
             <string>random</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QRadioButton" name="pf_source_hash">
            <property name="text">
             <string>source-hash</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QRadioButton" name="pf_round_robin">
            <property name="text">
             <string>round-robin</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="2">
        <spacer>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>81</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="1" rowspan="2">
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>130</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1" colspan="2">
        <widget class="QCheckBox" name="pf_static_port">
         <property name="text">
          <string>static-port</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="ASA8NATRuleOptions">
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Starting with v8.3 ASAs support NAT type &quot;static&quot; and &quot;dynamic&quot; for source NAT rules. Firewall Builder attempts to determine the correct type based on the information in the rule, but the calculated value can be overridden below.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string/>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QRadioButton" name="asa8_nat_auto">
            <property name="text">
             <string>Automatically detect NAT type &quot;static&quot; or &quot;dynamic&quot;. This rule is currently set to type &quot;%1&quot;</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QRadioButton" name="asa8_nat_dynamic">
            <property name="text">
             <string>Force rule to be NAT type &quot;dynamic&quot;. Note, rules with destination translation defined cannot be &quot;dynamic&quot;</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QRadioButton" name="asa8_nat_static">
            <property name="text">
             <string>Force rule to be NAT type &quot;static&quot;.</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QCheckBox" name="asa8_nat_dns">
         <property name="text">
          <string>Make this NAT rule translate DNS replies. You also need to enable DNS inspection in the firewall object advanced settings dialog.</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>265</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pf_bitmask</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pf_random</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pf_source_hash</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pf_round_robin</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pf_static_port</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pf_pool_type_none</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ipt_use_snat_instead_of_masq</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>307</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ipt_nat_random</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ipt_nat_persistent</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>474</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>asa8_nat_dns</sender>
   <signal>stateChanged(int)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>asa8_nat_auto</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>64</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>asa8_nat_dynamic</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>asa8_nat_static</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ipt_use_masq</sender>
   <signal>toggled(bool)</signal>
   <receiver>NATRuleOptionsDialog_q</receiver>
   <slot>changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changed()</slot>
 </slots>
</ui>
